hypershift-operator/controllers/nodepool/azure.go

```diff
@@ -34,6 +34,7 @@ func azureMachineTemplateSpec(hcluster *hyperv1.HostedCluster, nodePool *hyperv1
 		Identity:               capiazure.VMIdentityUserAssigned,
 		UserAssignedIdentities: []capiazure.UserAssignedIdentity{{ProviderID: hcluster.Spec.Platform.Azure.MachineIdentityID}},
 		SSHPublicKey:           sshKey,
+		FailureDomain:          failureDomain(nodePool),
```

Comment:
fwiw using `utilpointer.String(nodepool.Spec.Platform.Azure.AvailabilityZone)` directly should be fine there's no semantic difference for Azure provider between nil and ""